### PR TITLE
[12.0][MIG] storage_backend

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: storage_backend/README.rst [hunk 1]
```diff
@@ -1,48 +1,60 @@
...
+* Akretion
 
 Contributors
-------------
+~~~~~~~~~~~~
 
-* Sebastien Beau <sebastien.beau@akretion.com>
+* Sébastien BEAU <sebastien.beau@akretion.com>
 * Raphaël Reverdy <raphael.reverdy@akretion.com>
+* Florian da Costa <folrian.dacosta@akretion.com>
```
Comment (bealdav) [inline storage_backend/README.rst]: folrian oops

**Comment by florian-dacosta** [comment]
I think we talked about adding a dependency on server_environment, directly.
In all storage modules, instead of doing 1 module + 1 glue server_environment module each time...
Indeed, now that keychain module will be pluggued on server_environment, it is not a problem anymore...
@sebastienbeau

**Comment by florian-dacosta** [comment]
@rousseldenis What do you think about the server_environment dependency?
If I recall the we consider that most of pepole into the OCA already use server environment. For those you used also keychain (like akretion), it now depends on server_environment, so the dependency is not an issue.
For those who use nothing at all, server_environment won't block them as it store values in database by default...

**Comment by rousseldenis** [comment]
> @rousseldenis What do you think about the server_environment dependency?
> If I recall the we consider that most of pepole into the OCA already use server environment. For those you used also keychain (like akretion), it now depends on server_environment, so the dependency is not an issue.
> For those who use nothing at all, server_environment won't block them as it store values in database by default...

Hi Florian, I discussed about that with @lmignon and it is maybe more flexible to let it as is and add a new module 'storage_backend_environment' to implement the new way. If people want to implement another method, they are free to do it.

Maybe I will add that environment module in this migration flow if I have enough time.

What do you think ?

**Comment by florian-dacosta** [comment]
Well, yes, it would be more flexible of course, but I really doubt people will come with an other implementation, for storage, now that server environment is really flexible.
The sad thing is that we double the number of modules. (storage_backend_environment, storage_backend_s3_environment, storage_backend_sftp_environment, ...)
I would prefer to include this in the module, but I have not a strong opinion about this. And I am not even a user of these modules for now, so I leave people more involved in this to decide !

**Comment by lmignon** [comment]
@florian-dacosta I prefer to keep storage addons without dependencies on server_environement, In practice we decide to use server_environement by project and add extend the models for which we want to use it into the project.

**Review by lmignon — APPROVED** [review]

**Comment by simahawk** [comment]
> I prefer to keep storage addons without dependencies on server_environement, In practice we decide to use server_environement by project and add extend the models for which we want to use it into the project.

@lmignon is this still true even w/ the new version of server env? It allows you to edit settings via UI when you don't have params.
If we don't support this in the core we'll have a tons of *_env module for each specific storage_* module.

**Comment by lmignon** [comment]
@simahawk It's no more true with the new version of server env. We should adapt the code (even in shopinvader)

**Pull request merged**